packages/react-reconciler/src/ReactStrictModeWarnings.js

```diff
@@ -225,6 +232,15 @@ if (__DEV__) {
     fiber: Fiber,
     instance: any,
   ) => {
+    // Dedup strategy: Warn once per component.
+    // This is difficult to track any other way since component names
+    // are often vague and are likely to collide between 3rd party libraries.
+    // An expand property is probably okay to use here since it's DEV-only,
+    // and will only be set in the event of serious warnings.
+    if (didWarnAboutUnsafeLifecycles.has(fiber.type)) {
+      return;
+    }
+
```

Comment:
Why did these lines move above the `StrictRoot` ancestor check? I don't think they should have.

Reply:
I move it upwards intentionally. Because I was thinking that maybe we should not spare time to find its strict node if we already warned this fiber? in which case the `findStrictNode` becomes extraneous operations.

Tell me if I was wrong.

Reply:
Generally, I think error sanity checks (like the `StrictRoot` check) belong first in the function. In this case, I guess I agree that it doesn't matter practically speaking.

Your call 😄

Reply:
That makes sense to me now. I agree.